source/Fleet.cpp

```diff
@@ -204,21 +205,28 @@ void Fleet::Load(const DataNode &node)
 			{
 				resetVariants = false;
 				variants.clear();
-				total = 0;
 			}
-			variants.emplace_back(child);
-			total += variants.back().weight;
+			
+			int weight = 1;
+			string variantName;
+			if(child.Size() >= 2 + add && !child.IsNumber(1 + add))
+				variantName = child.Token(1 + add);
+			if(child.Size() >= 2 + add + !variantName.empty())
+				weight = child.Value(1 + add + !variantName.empty());
+			
+			if(!variantName.empty())
+				variants.emplace_back(GameData::Variants().Get(variantName), weight);
+			else
+				variants.emplace_back(child, weight);
```

Comment:
Would it be a good idea to store a `Variant` inside a fleet and then use that to load a `variant` node. Would avoid duplicating logic between a `Fleet` and a `Variant`.